hooks/useExchange.ts

```diff
@@ -33,6 +34,11 @@ type ExchangeCardProps = {
 type ExchangeModal = 'settle' | 'confirm' | 'approve' | 'redeem' | 'base-select' | 'quote-select';
 export type SwapRatio = 25 | 50 | 75 | 100;
 
+sdk.events.on('signer_connected', () => {
```

Comment:
This is a nice solution but I feel it would be good if we could keep the sdk interaction isolated to connector and redux and have state changes drive redux actions. So in this case the connector can be the source of truth for state of wallet connection, when a change happens it would update the sdk with `await setSigner` and then update state with connection info and status. I think then, a page component that is managing a data query can listen to a `useEffect` change on connection status / wallet address for triggering fetches, or maybe an `onConnected` callback we add to the connector.

And if the connector is awaiting setSigner before any state is updated hopefully there should be no need for listening to connected event.